src/eos-prepare-usb-update.c

```diff
@@ -329,7 +331,9 @@ pull_ready (GObject *object,
             GAsyncResult *res,
             gpointer pull_data_ptr)
 {
-  g_autoptr(EosPullData) pull_data = EOS_PULL_DATA (pull_data_ptr);
+  g_autoptr(EosPullData) pull_data = pull_data_ptr;
+
+  g_return_if_fail (EOS_IS_PULL_DATA (pull_data));
```

Comment:
Surely we want to run the `EOS_IS_PULL_DATA` check here before assigning the autoptr? (if the pointed-to object is not a `GObject`, calling `g_object_unref` on it will below up .... that said, we would have already hit programmer error in this case...)

Reply:
If the check fails, we’re hosed anyway.